anchor/message_sender/src/network.rs

```
@@ -0,0 +1,141 @@
+use crate::MessageSender;
+use database::{NetworkState, UniqueIndex};
+use openssl::error::ErrorStack;
+use openssl::hash::MessageDigest;
+use openssl::pkey::{PKey, Private};
+use openssl::rsa::Rsa;
+use openssl::sign::Signer;
+use ssv_types::consensus::UnsignedSSVMessage;
+use ssv_types::message::SignedSSVMessage;
+use ssv_types::msgid::DutyExecutor;
+use ssv_types::OperatorId;
+use ssz::Encode;
+use std::sync::Arc;
+use subnet_tracker::SubnetId;
+use tokio::sync::mpsc::error::TrySendError;
+use tokio::sync::{mpsc, watch};
+use tracing::{debug, error, warn};
+
+const SIGNER_NAME: &str = "message_sign_and_send";
+const SENDER_NAME: &str = "message_send";
+
+pub struct NetworkMessageSender {
+    processor: processor::Senders,
+    network_tx: mpsc::Sender<(SubnetId, Vec<u8>)>,
+    private_key: PKey<Private>,
+    database: watch::Receiver<NetworkState>,
+    operator_id: OperatorId,
+    subnet_count: usize,
+}
+
+impl MessageSender for Arc<NetworkMessageSender> {
+    fn sign_and_send(&self, message: UnsignedSSVMessage) {
+        let sender = self.clone();
+        self.processor
+            .urgent_consensus
+            .send_blocking(
+                move || {
+                    let signature = match sender.sign(&message) {
+                        Ok(signature) => signature,
+                        Err(err) => {
```

Do you know if we need to propagate this error?

---

This is one of those errors that really really should never happen, and where there is not really a way to fall back. So there is no reason to propagate it somewhere.

---

According to Murphy's law: Anything that can go wrong will go wrong :) It might be useful to test what happens to the rest of the system when this happens. I'll keep this in mind in my tasks.